watcher/watcher.go

```diff
@@ -25,35 +25,90 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type BurnLogResult struct {
+	Result ethereumbindings.MintGatewayLogicV1LogBurn
+	Error  error
+}
+
+type BurnLogFetcher interface {
+	FetchBurnLogs(ctx context.Context, from uint64, to uint64) (chan BurnLogResult, error)
+}
+
+type EthBurnLogFetcher struct {
+	bindings *ethereumbindings.MintGatewayLogicV1
+}
+
+func NewBurnLogFetcher(bindings *ethereumbindings.MintGatewayLogicV1) EthBurnLogFetcher {
+	return EthBurnLogFetcher{
+		bindings: bindings,
+	}
+}
+
+// This will fetch the burn event logs using the ethereum bindings and emit them via a channel
+// We do this so that we can unit test the log handling without calling ethereum
+func (fetcher EthBurnLogFetcher) FetchBurnLogs(ctx context.Context, from uint64, to uint64) (chan BurnLogResult, error) {
+	iter, err := fetcher.bindings.FilterLogBurn(
+		&bind.FilterOpts{
+			Context: ctx,
+			Start:   from,
+			End:     &to,
+		},
+		nil,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	resultChan := make(chan BurnLogResult)
+
+	go func() {
+		for iter.Next() {
+			err := iter.Error()
+			if err != nil {
+				resultChan <- BurnLogResult{Error: err}
+				break
+			}
+			resultChan <- BurnLogResult{Result: *iter.Event}
+			select {
+			case <-ctx.Done():
+				break
```

Comment:
This `break` only breaks the `select` statement, not the `for` loop.

Reply:
Good catch